packages/jest-cli/src/reporters/CoverageReporter.js

```diff
@@ -59,6 +62,7 @@ class CoverageReporter extends BaseReporter {
     runnerContext: RunnerContext,
   ) {
     this._addUntestedFiles(config, runnerContext);
+    const { map, sourceFinder } = this._sourceMapStore.transformCoverage(this._coverageMap);
```

Comment:
@dmitriiabramov istanbul-lib-source-maps has code that plucks the source map from the coverage object as part of this transformCoverage call. If no source map is tacked onto the object, it's not mapped.

@cpojer Will fix shortly!